test/OpenTelemetry.Collector.Dependencies.Tests/SqlClientTests.cs

```diff
@@ -0,0 +1,202 @@
+﻿// <copyright file="SqlClientTests.cs" company="OpenTelemetry Authors">
+// Copyright 2018, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Data;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.Data.SqlClient;
+using Moq;
+using OpenTelemetry.Collector.Dependencies.Implementation;
+using OpenTelemetry.Trace;
+using OpenTelemetry.Trace.Configuration;
+using OpenTelemetry.Trace.Export;
+using Xunit;
+
+namespace OpenTelemetry.Collector.Dependencies.Tests
+{
+    public class SqlClientTests : IDisposable
+    {
+        private const string TestConnectionString = "Data Source=(localdb)\\MSSQLLocalDB;Database=master";
+
+        private readonly FakeSqlClientDiagnosticSource fakeSqlClientDiagnosticSource;
+
+        public SqlClientTests()
+        {
+            this.fakeSqlClientDiagnosticSource = new FakeSqlClientDiagnosticSource();
+        }
+
+        public void Dispose()
+        {
+            this.fakeSqlClientDiagnosticSource.Dispose();
+        }
+
+        [Theory]
+        [InlineData(SqlClientDiagnosticListener.SqlDataBeforeExecuteCommand, SqlClientDiagnosticListener.SqlDataAfterExecuteCommand)]
+        [InlineData(SqlClientDiagnosticListener.SqlMicrosoftBeforeExecuteCommand, SqlClientDiagnosticListener.SqlMicrosoftAfterExecuteCommand)]
+        public void SqlClientCallsAreCollectedSuccessfully(string beforeCommand, string afterCommand)
+        {
+            var activity = new Activity("Current").AddBaggage("Stuff", "123");
+            activity.Start();
+
+            var spanProcessor = new Mock<SpanProcessor>();
+            var tracer = TracerFactory.Create(b => b
+                    .AddProcessorPipeline(p => p.AddProcessor(_ => spanProcessor.Object)))
+                .GetTracer(null);
+
+            using (new SqlClientCollector(tracer))
+            {
+                var operationId = Guid.NewGuid();
+                var sqlConnection = new SqlConnection(TestConnectionString);
+                var sqlCommand = sqlConnection.CreateCommand();
+                sqlCommand.CommandText = "SP_GetOrders";
+                sqlCommand.CommandType = CommandType.StoredProcedure;
```

Comment:
could you add a test with CommandType as Text as well?

Reply:
Done